Source/Evolve/WarpXEvolve.cpp

```diff
@@ -83,9 +83,9 @@ WarpX::Evolve (int numsteps)
         if (verbose) {
             amrex::Print() << "\nSTEP " << step+1 << " starts ...\n";
         }
-        if (warpx_py_beforestep) {
+        if ( warpx_callback_py_map.count("beforestep") == 1u ) {
```

Comment:
This `if` block pattern appears many times. Could this be put into a subroutine, something like

```
void call_python_callback(std::string name) {
    if (warpx_callback_py_map.count(name) == 1u) {
        WARPX_PROFILE("warpx_py_"+name);
        warpx_callback_py_map[name]();
    }
}
```

Then this would be
`call_python_callback("beforestep");`

Reply:
Great suggestion. Thanks!